WordPress/gutenberg
# Sidebar Tabs: Add text only mode for block inspector tabs

Fixes: https://github.com/WordPress/gutenberg/issues/46871

## What?

Display textual labels for the Block Inspector tabs when the "Show button text labels" preference is set.

## Why?
It brings the inspector tabs more in line with the main editor and block toolbars when the "Show button text labels" option is set.

## How?

Uses a similar CSS approach to the main and block toolbars. The difference is that the `aria-label` text is set as the content on the `::before` pseudo element as `::after` is already used for styling a TabPanel's active tab.

## Testing Instructions

1. Enable the "Block Inspector Tabs" Gutenberg experiment
2. Edit a post, adding a paragraph and navigation blocks
3. Select each block and ensure the tabs display correctly with icons
4. From the top right "more menu", select Preferences
5. Toggle on the "Show button text labels" preference
6. Close the more menu and ensure a block is still selected
7. Check that the tabs now display the textual label for the button and the icon is hidden


### Testing Instructions for Keyboard

1. Enable the "Block Inspector Tabs" Gutenberg experiment
2. Edit a post and add a paragraph block
3. Tab through into the block inspector sidebar and ensure tabs are present and still navigable with the arrow keys
4. Add a navigation block and tab into the sidebar again, checking the tabs work
5. Open the "options" menu from the main editor toolbar. If in the block inspector you can navigate there by using the previous editor part keyboard shortcut twice e.g. <kbd>⌃</kbd> + <kbd>⇧</kbd> + <kbd>`</kbd>
6. Select the Preferences option
7. Within the preferences modal tab through to the "Show button text labels" preference and toggle it on
8. Escape out of the preferences modal and the options menu.
9. Reselect a block if necessary
10. Tab through to the block inspector again and check that the tabs no longer display the icon and instead show the textual label

## Screenshots or screencast 

<img width="253" alt="Screenshot 2023-01-10 at 2 25 06 pm" src="https://user-images.githubusercontent.com/60436221/211464765-e47f177f-f209-48c4-8eb4-2fd27420de74.png">
